pkg/applications/installer.go

```diff
@@ -42,14 +44,39 @@ type ApplicationInstaller interface {
 
 // ApplicationManager handles the installation / uninstallation of an Application on the user-cluster.
 type ApplicationManager struct {
+	// Namespace where credentials secrets are stored.
+	SecretNamespace string
 }
 
 // Apply creates the namespace where the application will be installed (if necessary) and installs the application.
 func (a *ApplicationManager) Apply(ctx context.Context, log *zap.SugaredLogger, seedClient ctrlruntimeclient.Client, userClient ctrlruntimeclient.Client, applicationInstallation *appskubermaticv1.ApplicationInstallation) error {
+	// initialize tools
+	sourceProvider, err := providers.NewSourceProvider(ctx, seedClient, &applicationInstallation.Status.ApplicationVersion.Template.Source, a.SecretNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to initialize source provider: %w", err)
+	}
+
+	downloadDest, err := os.MkdirTemp("", applicationInstallation.Namespace+"-"+applicationInstallation.Namespace)
```

Comment:
This can potentially create HUGE tmp directories depending on the git source. We need to put in safeguards, like using an empty dir in the user-cluster Deployment, and limiting the amount of disk space this can eat. Otherwise you can fully consume the underlying node's disk.

Reply:
(I know there's a shallow clone happening but IMHO we should still make sure this gets its own temporary volume with storage limits on it)

Reply:
you're totally right, i forgot we don't have default limits :s  working on it.

Reply:
@embik 
I have added an empty dir with a default size is 300Mi.

To choose this default value, I take a look at the size of 3 repositories:
* https://github.com/bitnami/charts: set of ~100 charts provided by bitnami
* https://github.com/prometheus-community/helm-charts: contains the different charts for deploying prometheus stack
* https://github.com/kubermatic/kubermatic

I choose these repositories because the first 2 serve only charts and the last one serves charts and code (with history).

|   repository         | clone all repo   | clone a tag  |
|-----------------|----------------|-------------|
| bitanmi               | 199M                 | 38M             |
| prometheus       | 87M                   | 6,7M             |
| kubermatic.       | 191M                  | 50M             |

So a 300Mi cache should be more than enough to download sources from a repo and handle dependencies.

_note: the max reconciliation of the application controller is 1._